extensions/wikia/Chat2/Chat.class.php

```diff
@@ -69,26 +72,34 @@ static public function banUser($userNameToKickBan, $kickingUser, $time, $reason)
 			'reason' => $reason,
 		] );
 
-		global $wgCityId;
-		wfProfileIn( __METHOD__ );
-		$errorMsg = "";
-		$PERMISSION_TO_KICKBAN = "chatmoderator";
-		$userToKickBan = User::newFromName($userNameToKickBan);
+		$kickBanPermission = 'chatmoderator';
+		$userToKickBan = User::newFromName( $userNameToKickBan );
 
-		if( ($userToKickBan instanceof User) && $kickingUser->isAllowed( $PERMISSION_TO_KICKBAN ) ){
+		// Make sure user doing the kick/ban has permission to do so
+		if ( !( $userToKickBan instanceof User ) ||
+			 !$kickingUser->isAllowed( $kickBanPermission ) ) {
 
-			if( $userToKickBan->isAllowed( $PERMISSION_TO_KICKBAN ) && !$kickingUser->isAllowed('chatstaff') && !$kickingUser->isAllowed('chatadmin') ){
-				$errorMsg .= wfMsg('chat-ban-cant-ban-moderator')."\n";
-			} else {
-				self::banUserDB($wgCityId, $userToKickBan, $kickingUser, $time, $reason, $time == 0 ?  'remove':'add' );
-			}
-		} else {
-			$errorMsg .= wfMsg('chat-ban-you-need-permission', $PERMISSION_TO_KICKBAN)."\n";
+			return wfMessage( 'chat-ban-you-need-permission', $kickBanPermission )
+				->inContentLanguage()->text() . "\n";
 		}
 
-		wfProfileOut( __METHOD__ );
-		return ( $errorMsg=="" ? true : $errorMsg);
-	} // end banUser()
+		// Make sure we aren't trying to kick/ban someone who shouldn't be kick/banned
+		if ( $userToKickBan->isAllowed( $kickBanPermission ) && !$kickingUser->isAllowed( 'chatstaff' ) && !$kickingUser->isAllowed( 'chatadmin' ) ) {
+			return wfMessage( 'chat-ban-cant-ban-moderator' )
+				->inContentLanguage()->text() . "\n";
```

Comment:
None of this was escaped before, it doesn't contain any user supplied data and some of these things go to a logging table so I'm inclined to leave it as is and not introduce any possible escaping issues.